ParcelInvoiceMaker/ParcelInvoiceMaker/Processor/Parcel/ParcelProcessor.swift

```diff
@@ -23,11 +27,18 @@ final class ParcelOrderProcessorImpl: ParcelOrderProcessor {
 extension ParcelOrderProcessorImpl {
     
     // 택배 주문 처리 로직
-    func process(parcelInformation: ParcelInformation, onComplete: (ParcelInformation) -> Void) {
+    func process(
+        parcelInformation: ParcelInformation,
+        receiptInfo: ReceiptProvideInfo,
+        onComplete: (ParcelInformation) -> Void
+    ) {
         
         // 데이터베이스에 주문 저장
         databaseParcelInformationPersistence.save(parcelInformation: parcelInformation)
         
+        // 영수증 전송
+        receiptInfo.sendReceipt()
+        
         onComplete(parcelInformation)
```

Comment:
```suggestion
     func process(parcelInformation: ParcelInformation, receiptInfo: ReceiptProvideInfo) async -> ParcelInformation {
        
        // 데이터베이스에 주문 저장
        databaseParcelInformationPersistence.save(parcelInformation: parcelInformation)
        
        // 영수증 전송, 작업이 완료될 때까지 기다림
        await receiptInfo.sendReceipt()
        
        return parcelInformation
    }
```
컨커런시를 사용하면서 콜백 메소드를 받는게 조금 부자연스러운것 같은데,
onComplete 핸들러를 사용하지 않고 위처럼 표현하는건 어떠신가요? 
작성하신 코드에서 onComplete는 sendReceipt의 완료와 상관없이 호출되고 있기도 하구요.
만약 제가 제안드린 것이 괜찮다면, 관련해서 sendRecipt쪽도 수정이 되어야 할 것같아요.

Reply:
말씀해주신 내용이 컨커런시를 자연스럽게 사용하는 방법이라는 것에 동의합니다 🙂
다음 푸시에 반영해볼게요!